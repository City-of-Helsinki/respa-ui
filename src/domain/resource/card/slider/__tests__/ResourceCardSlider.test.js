import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { UnconnectedResourceCardSlider } from '../ResourceCardSlider';
import Resource from '../../../../../../app/utils/fixtures/Resource';
import Unit from '../../../../../../app/utils/fixtures/Unit';


describe('ResourceCardSlider', () => {
  test('renders correctly', () => {
    const props = {
      history: {},
      location: {},
      resources: [Resource.build()],
      unit: Unit.build(),
      date: '30-07-2019',
      onFavoriteClick: jest.fn(),
      onFilterClick: jest.fn()
    };
    const wrapper = shallow(
      <UnconnectedResourceCardSlider {...props} />
    );

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
