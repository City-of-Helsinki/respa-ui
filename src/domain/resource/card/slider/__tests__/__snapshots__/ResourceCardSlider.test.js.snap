// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceCardSlider renders correctly 1`] = `
<div
  className="app-ResourceCardSlider"
>
  <InjectIntl(Connect(InjectT(ResourceCard)))
    date="30-07-2019"
    history={Object {}}
    isStacked={false}
    location={Object {}}
    onFavoriteClick={[MockFunction]}
    onFilterClick={[MockFunction]}
    resource={
      Object {
        "availableHours": Array [],
        "equipment": Array [],
        "id": "r-1",
        "isFavorite": false,
        "name": "Resource-1",
        "needManualConfirmation": false,
        "openingHours": Array [],
        "products": Array [],
        "public": true,
        "requiredReservationExtraFields": Array [],
        "reservable": true,
        "reservableAfter": null,
        "slotSize": "00:30:00",
        "supportedReservationExtraFields": Array [],
        "unit": "u-1",
        "userPermissions": Object {
          "canMakeReservations": true,
          "isAdmin": false,
        },
      }
    }
    unit={
      Object {
        "id": "u-1",
        "name": "Unit-1",
      }
    }
  />
</div>
`;
